src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/AceEditorBackgroundLinkHighlighter.java

```diff
@@ -736,34 +778,6 @@ interface Styles extends CssResource
 
    // Private Members ----
 
-   private static class MarkerRenderer extends JavaScriptObject
-   {
-      protected MarkerRenderer() {}
-
-      public static final native MarkerRenderer create(final AceEditorNative editor,
-                                                       final String clazz,
-                                                       final String title)
-      /*-{
-         var markerBack = editor.renderer.$markerBack;
-         return $entry(function(html, range, left, top, config) {
-            // HACK: we take advantage of an implementation detail of
-            // Ace's 'drawTextMarker' implementation. Ace constructs
-            // HTML for the generated markers with code of the form:
-            //
-            //    html = "<div style='..." + extraStyle + "'>"
-            //
-            // We take advantage of this, and inject our 'extraStyle'
-            // to close the style attribute we were intended to be
-            // locked in, and instead inject a 'title' attribute instead.
-            var extra = "' title='" + title;
-            if (range.isMultiLine())
```

Comment:
This hack did not work anymore. At the end of the day, this ended up in `this.elt` down in ace : 

```javascript
    this.elt = function(className, css) {
        var x = this.i != -1 && this.element.childNodes[this.i];
        if (!x) {
            x = document.createElement("div");
            this.element.appendChild(x);
            this.i = -1;
        } else {
            this.i++;
        }
        x.style.cssText = css;
        x.className = className;
    };
```

and specifically ` x.style.cssText = css;` can't be fooled. We can probably think of another way to add the `title`

Reply:
Also, because this was using `.$markerBack` the markers were removed e.g. when highlight markers were added.

Reply:
Can you clarify what you want to do here? IIUC the Ace marker API should already have some tooling for providing hover text. Or do we need an Ace patch to make this more straightforward?

Reply:
I've used a different hack based on the `this.i` in `this.elt` that ends up with `x.setAttribute("title", title)`

Reply:
<img width="383" alt="image" src="https://user-images.githubusercontent.com/2625526/186628941-b8de09f4-c780-42f9-9d6f-6041ba0f2a74.png">

Reply:
Another way to deal with this could be to do something like what is done for lint markers, via the `DiagnosticsBackgroundPopup` class.